File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>

int main(int argc, string argv[])
{
    if (argc != 2 || !((int) argv[1][0] >= 48 
                       && (int) argv[1][0] <= 57)) //checking number of arguements and first character of the arguement 1 
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    else
    {
        int key, ch;
        // extracting int key from string arguement
        key = atoi(argv[1]) % 26; // making key % 26
        string plain_text;
        plain_text = get_string("plaintext:  ");
        for (ch = 0; plain_text[ch] != '\0'; ch++)
        {
            if (plain_text[ch] >= 65 && plain_text[ch] <= 90) 
            {
                plain_text[ch] = ((plain_text[ch] - 'A' + key) % 26) + 'A';  
                // adding key value to the character 
            }
            else if (plain_text[ch] >= 97 && plain_text[ch] <= 122)
            {
                plain_text[ch] = ((plain_text[ch] - 'a' + key) % 26) + 'a';
                // adding key value to the character
            }
                
        }
        printf("ciphertext: %s\n", plain_text); // printing cipher_text
        return 0;
    }
}

File: pset6/caesar/caesar.py
import sys
from cs50 import get_string


def main():
    if len(sys.argv) != 2:
        print("Usage: ./caesar k")
        sys.exit(1)

    k = int(sys.argv[1])
    plaintext = get_string("plaintext: ")

    print("ciphertext: ", end="")

    for ch in plaintext:
        if not ch.isalpha():
            print(ch, end="")  # printing non alpha characters as it is
            continue

        ascii_offset = 65 if ch.isupper() else 97  # ascii offset calculation for lower case letters and upper case letters

        pi = ord(ch) - ascii_offset  # getting value of ascii character in between 0 and 25
        ci = (pi + k) % 26  # adding k to the digit and modulus by 26

        print(chr(ci + ascii_offset), end="")  # printing character of encrypted text

    print()

    return 0


if __name__ == "__main__":
    main()
